### PR TITLE
Add a `gen_random_text_uuid` scalar

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

Closes https://github.com/crate/crate/issues/11039

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/expression/scalar/GenRandomTextUUIDFunctionTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import io.crate.testing.MoreMatchers;
+
+public class GenRandomTextUUIDFunctionTest extends AbstractScalarFunctionsTest {
+
+    @Test
+    public void test_gen_random_text_uuid_returns_a_uuid() throws Exception {
+        assertEvaluate(
+            "gen_random_text_uuid()",
+            MoreMatchers.withFeature((String x) -> x.length(), "length", Matchers.is(20)));
+    }
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/expression/scalar/GenRandomTextUUIDFunctionTest.java:36]: Could probably also use a fixed see to make the evaluation deterministic. WDYT?

Comment (seut) [reply server/src/test/java/io/crate/expression/scalar/GenRandomTextUUIDFunctionTest.java:36]: Yes I think that would be valuable.

**Review by seut — APPROVED** [review]
👍 Left a comment related to a deterministic value-comparable test, but feel free to decide if its worth to do this now.

File: docs/general/builtins/scalar.rst [hunk 2]
```diff
@@ -1723,6 +1723,18 @@ Returns: ``double precision``
 
     Every call to ``random`` will yield a new random number.
 
+
+.. _scalar-gen_random_text_uuid:
+
+``gen_random_text_uuid()``
+--------------------------
+
+Returns a random time based UUID as ``text``. The returned id is similar to
```
Comment (norosa) [inline docs/general/builtins/scalar.rst:1732]:
```suggestion
Returns a random time based UUID as ``text``. The returned ID is similar to
```

File: docs/general/builtins/scalar.rst [hunk 3]
```diff
@@ -1723,6 +1723,18 @@ Returns: ``double precision``
...
+flake IDs and well suited for use as primary key value.
+
+Note that the id is opaque and the implementation is free to change.
```
Comment (norosa) [inline docs/general/builtins/scalar.rst:1735]:
```suggestion
Note that the ID is opaque (i.e., not to be considered meaningful in any way)
and the internal implementation may change in the future.
```

File: server/src/test/java/io/crate/expression/scalar/GenRandomTextUUIDFunctionTest.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/test/java/io/crate/expression/scalar/GenRandomTextUUIDFunctionTest.java:36]: Oh, forgot about the time component in the uuid. Making it deterministic is not possible without also using some overriden `TimeBasedUUIDGenerator`. I'll leave it as it is.

**Review by norosa — COMMENTED** [review]
lgtm

**Pull request merged**